pkg/fetcher/fetcher.go

```diff
@@ -414,6 +429,58 @@ func (u *URLFetcher) IsNonretryableClientError() bool {
 		s != http.StatusLocked && s != http.StatusTooManyRequests
 }
 
+// Build error message for unexpected http code (nonretryable client errors and all other errors)
+// extract message details from response body stream if there is one
+func (u *URLFetcher) buildRegistryErrMsg(httpStatusCode int, url *url.URL, respBody io.ReadCloser) string {
+	errMsg := fmt.Sprintf("Unexpected http code: %d (%s), URL: %s", httpStatusCode, http.StatusText(httpStatusCode), url)
+
+	errDetail, err := u.extractErrResponseMessage(respBody)
+	if err != nil {
+		return errMsg
+	}
+
+	errMsg += fmt.Sprintf(", Message: %s", errDetail)
+
+	return errMsg
+}
+
+// Extract message from error response body (#5951)
+// The error response body is assumed to follow the convention of Docker
```

Comment:
`The error response body is assumed to follow the convention of Docker` is redundant since the comment for `RegistryErrorRespBody` includes this.